### DISCUSSION TITLE
curl example for filtering stake results by, policy_id and asset_name

### DISCUSSION THREAD

**Discussion by russell-rob** [CardanoSolutions/kupo#52] [Q&A] [answered]
hi i'm have trouble building a curl to filter results by policy_id and asset_name ;

this works;
curl http://localhost:1442/matches/stake1uyj83ggpzh227rh6gjq06avdjfl4x4n28rq29xrj8rkfx0c2wpcwt?unspent

i want filter this by a policy and asset 
this doesn't work
curl http://localhost:1442/matches/stake1uyj83ggpzh227rh6gjq06avdjfl4x4n28rq29xrj8rkfx0c2wpcwt?unspent&policy_id=efff17a2e610e5c9f46e33fe964c39602399f95aa367b37f039c1ef&asset_name=43617264616e6f466f725468654d6173736573

**Comment by KtorZ** [comment, marked as answer]
Hey, that decoded policy id is invalid. It's only 27-byte long whereas a policy id is always exactly 28 bytes. When base16-encoded, the strings is twice as long. The string you've passed is 55 characters so you're missing exactly one character, likely at the beginning or the end from a wrong copy-pasting.

**Reply by russell-rob** [reply]
that solved it, thank you.

this works;
curl 'http://localhost:1442/matches/stake1uyj83ggpzh227rh6gjq06avdjfl4x4n28rq29xrj8rkfx0c2wpcwt?unspent&policy_id=efff17a2e610e5c9f46e33fe964c39602399f95aa367b37f039c1efc&asset_name=43617264616e6f466f725468654d6173736573'
